docs/imaging/sync.rst

```diff
@@ -4,31 +4,117 @@
+
+  Setup.Sync & {"setup_name":"Mini Blue", "setup_timestamp":"2020-08-20 00:00:00"}
+
+.. figure:: /_static/imaging/sync/setup-sync-table.png
+   :alt: Contents of the ``Setup.Sync`` table
+
+
+When a new Session is ingested, the synchronisation data is inserted into the ``Sync`` table and its part tables. Each Session has one (or more) entries in the ``Sync`` table, corresponding to a single time series, from a single instrument, with each time series being identified by a ``sync_name``.
+
+The actual timestamps themselves are stored in the column ``sync_data`` of the ``Sync`` table. The values are stored in *samples*. The sample rate (samples per second) is also stored, in column ``sample_rate``, if you wish to convert into time periods in seconds. 
+
+
+Sync Names and Generic Names
+--------------------------------
+
+The ``Setup.Sync`` entries have two related attributes:
+
+* ``sync_name``
+* ``generic_name``
+
+The most important attribute here is the ``generic_name``. This defines how further downstream tables identfy *which* Sync stream they should be looking for, to pair time-series data with its timestamps. The main ``generic_names`` are as follows:
+
+* ``frames_imaging`` : timestamps for two-photon imaging frames
+* ``tracking_rotary`` : timestamps of 1-dimensional, wheel-encoder data
+* ``Tracking2LED`` : timestamps of video frames of open-field tracking, where the subject is identified by a pair of head-mounted LEDs
+* ``TrackingDLC`` : timestamps of video frames intended to be analysed with DLC (see :ref:`Imaging dlc`).
+
+``sync_name`` is intended to provide a name that conveys more information to the users.
+
+
+In most cases, users should not need to extract data from the ``Sync`` tables directly: timestamp information is inserted into the more common tables (``Tracking.Linear``, ``Tracking.OpenField``, ``SignalTracking``).
+
+The following example is a simplifed approach to fetching Synchronisation data either where it is not already present, or where the user may need to apply some processing to it first.
+
+.. code-block:: python
+    
+    # Use an arbitrary key to the `Cell.Traces` table
+    key = {"session_name":"d6ba671faad74695", "cell_id":3, "channel":"primary"}
+    df_f = (Cell.Traces & key).fetch1("df_f")
+
+    # Identify what Setup was in use
+    setup = (Session * MetaSession.Setup & key).fetch1()
+
+    # Identify what sync name we should be looking for
+    generic_name = "frames_imaging" # cell.Traces data -> it's from the 2p data
+    sync_name = (Setup.Sync & setup & {"generic_name":generic_name}).fetch1("sync_name")
+
+    # Get the sync data and sampling rate, and convert from samples to seconds
+    sync_data_samples, fs = (Sync & key & {"sync_name":sync_name}).fetch1("sync_data", "sample_rate")
+    sync_data_s = sync_data_samples / fs
+
+    # Plot the resulting data
+    fig, ax = plt.subplots()
+    ax.plot(sync_data_s, df_f)
+    ax.set_ylabel("df/f")
+    ax.set_xlabel("time (s)")
```

Comment:
Some time ago I wrote a helper for this https://github.com/kavli-ntnu/dj-moser-imaging/blob/de303b0bf293a59c78280547f88c4de66b749a8a/dj_schemas/utils.py#L529 

It might be good to provide this function as part of the user_imaging repo (future goal?).